sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -316,7 +323,7 @@ object JoinReorderDP extends PredicateHelper with Logging {
   }
 
   /** Map[set of item ids, join plan for these items] */
-  type JoinPlanMap = Map[Set[Int], JoinPlan]
+  type JoinPlanMap = mutable.LinkedHashMap[Set[Int], JoinPlan]
```

Comment:
I think this is OK, though it's not strictly necessary to make the type def this specific

Reply:
> Change to use LinkedHashMap instead of Map to store foundPlans in JoinReorderDP.search method to ensure same iteration order with same insert order because iteration order of Map behave differently under Scala 2.12 and 2.13

Can we make it as a separate PR? The plan changes need more reviews

Reply:
Do you mean to use `Map[Set[Int], JoinPlan]` or not to define `JoinPlanMap` type?

Reply:
@gatorsmile Use a separate JIRA number?

Reply:
Same JIRA I think, just a separate PR. I'm not sure it matters that much though, it's a tiny part of the change really? we just need more eyes on the plan changes.

Reply:
> Same JIRA I think, just a separate PR. I'm not sure it matters that much though, it's a tiny part of the change really? we just need more eyes on the plan changes.

It seems that there is no way to divide 2 PR because if we change `CostBasedJoinReorder `  only,  the test cases in `sql/core` module will be failed.

Reply:
So all excepted plan changes are due to the use of `LinkedHashMap` instead of `Map`